src/connection/connection_support.c

```diff
@@ -2945,3 +2957,204 @@ css_conn_entry::init_pending_request ()
   pending_request_count = 0;
 }
 // *INDENT-ON*
+
+CSS_CONN_ENTRY *
+css_connect_to_master_server (int master_port_id, const char *message_to_master, int message_to_master_length)
+{
+  char hname[CUB_MAXHOSTNAMELEN];
+  CSS_CONN_ENTRY *conn;
+  unsigned short rid;
+  int response, response_buff;
+  int server_port_id;
+  int connection_protocol;
+#if !defined(WINDOWS)
+  std::string pname;
+  int datagram_fd, socket_fd;
+#endif
+
+  css_Service_id = master_port_id;
+  if (GETHOSTNAME (hname, CUB_MAXHOSTNAMELEN) != 0)
+    {
+      return NULL;
+    }
+  conn = css_make_conn (0);
+  if (conn == NULL)
+    {
+      er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_ERROR_DURING_SERVER_CONNECT, 1, message_to_master);
+      return NULL;
+    }
+
+  /* select the connection protocol */
+  if (css_Server_use_new_connection_protocol)
+    {
+      // Windows
+      connection_protocol = SERVER_REQUEST_NEW;
+    }
+  else
+    {
+      // Linux and Unix
+      connection_protocol = SERVER_REQUEST;
+    }
+
+  if (css_common_connect
+      (conn, &rid, hname, connection_protocol, message_to_master, message_to_master_length, master_port_id, 0,
+       true) == NULL)
+    {
+      goto fail_end;
+    }
+
+  if (css_readn (conn->fd, (char *) &response_buff, sizeof (int), -1) != sizeof (int))
+    {
+      goto fail_end;
+    }
+
+  response = ntohl (response_buff);
+
+  TRACE ("css_connect_to_master_server received %d as response from master\n", response);
+
+  switch (response)
+    {
+    case SERVER_ALREADY_EXISTS:
+#if !defined (SERVER_MODE)
+      if (IS_MASTER_CONN_NAME_HA_COPYLOG (message_to_master))
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_COPYLOG_ALREADY_EXISTS, 1,
+		  GET_REAL_MASTER_CONN_NAME (message_to_master));
+	}
+      else if (IS_MASTER_CONN_NAME_HA_APPLYLOG (message_to_master))
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_APPLYLOG_ALREADY_EXISTS, 1,
+		  GET_REAL_MASTER_CONN_NAME (message_to_master));
+	}
+      else if (IS_MASTER_CONN_NAME_HA_SERVER (message_to_master))
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_SERVER_ALREADY_EXISTS, 1,
+		  GET_REAL_MASTER_CONN_NAME (message_to_master));
+	}
+      else
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_SERVER_ALREADY_EXISTS, 1, message_to_master);
+	}
+#else // SERVER_MODE
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_SERVER_ALREADY_EXISTS, 1, message_to_master);
+#endif // SERVER_MODE
```

Comment:
shorter, but more ambiguous :)
```suggestion
      else
#endif // !SERVER_MODE
	{
	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_SERVER_ALREADY_EXISTS, 1, message_to_master);
	}
```

Reply:
Changed. Now is even shorter and very clear